watched_keywords.txt

```diff
@@ -3339,7 +3339,7 @@
 1525903476	Zoe	gamaplus\.ir
 1525908727	Byte Commander	seoexpertspk\.com
 1525911965	DJMcMayhem	metropollitan\.com
-1525925733	Makyen	f+u+[ck]+(?<!fuck)(?<!fuk)
+1525925733	Makyen	(?:(?:m(?:[^A-Za-z]|&\w+;)*)+(?:o(?:[^A-Za-z]|&\w+;)*)+(?:t(?:[^A-Za-z]|&\w+;)*)+(?:h(?:[^A-Za-z]|&\w+;)*)+(?:e(?:[^A-Za-z]|&\w+;)*)*(?:r(?:[^A-Za-z]|&\w+;)*)+)?(?:f(?:[^A-Za-z]|&\w+;)*)+(?:u(?:[^A-Za-z]|&\w+;)*)+(?:(?:[^A-Za-z]|&\w+;)*[ck])+(?<!fuck)(?<!fuk)(?:[^A-Za-z]|&\w+;)*(?:(?:i(?:[^A-Za-z]|&\w+;)*)+(?:n(?:[^A-Za-z]|&\w+;)*)+(?:g(?:[^A-Za-z]|&\w+;)*)+|(?:e(?:[^A-Za-z]|&\w+;)*)+(?:[rd](?:[^A-Za-z]|&\w+;)*)+)?(?:s(?:[^A-Za-z]|&\w+;)*)*
```

Comment:
So... I can read a regex and know what each bit does, but run me through the theory behind what this is intended to do?

Reply:
This detects "fuck", or "motherfuck" (er|ed|ing)?s? when it's interspersed with either repeated characters or non-alpha characters, either as unicode characters or HTML entities. Due to the negative look-behind it does not detect the simple cases which are already detected using other regexes. Basically, the intent is to catch versions of this word which will be readable by a human, but which diverge significantly from what's normally detected.

Here's a [brief regex101 test for it](https://regex101.com/r/qojS0K/6/).